trino-lb/src/maintenance/query_count_fetcher.rs

```diff
@@ -152,8 +148,40 @@ impl QueryCountFetcher {
         }
     }
 
-    #[instrument(skip(self))]
-    async fn process_cluster(&self, cluster: &TrinoClusterConfig) {
+    /// Update the query counter for the given cluster.
+    ///
+    /// In case the cluster should be reachable, we fetch the current counter and store that.
+    /// In case we know there is no point in fetching the counter (as e.g. the cluster is turned off), we store a query
+    /// count of zero (0) to avoid having dangling clusters with non-zero query counts (when in fact they are not
+    /// executing any query at all).
```

Comment:
suggestion: Use bullet points, slight reword.

```suggestion
    /// Update the query count for the given cluster.
    ///
    /// - In case the cluster is reachable, fetch the current query count and store it.
    /// - In case the cluster is in a state in which it cannot process queries (e.g. the cluster is
    ///   turned off or currently starting), store a query count of zero (0) to avoid dangling clusters
    ///   with non-zero query counts because they are not executing any query.
```

Reply:
Please have a look at 13f89b9